lib/src/chart/line_chart/line_chart.dart

```diff
@@ -82,31 +85,46 @@ class _LineChartState extends AnimatedWidgetBaseState<LineChart> {
           touchData.touchCallback(response);
         }
       },
-      onPanCancel: () {
+      onTapCancel: () {
         final Size chartSize = _getChartSize();
         if (chartSize == null) {
           return;
         }
 
-        final LineTouchResponse response = _touchHandler?.handleTouch(
-            FlPanEnd(Offset.zero, Velocity(pixelsPerSecond: Offset.zero)), chartSize);
+        final LineTouchResponse response =
+            _touchHandler?.handleTouch(FlTap(_lastTouchedPosition), chartSize);
         if (_canHandleTouch(response, touchData)) {
           touchData.touchCallback(response);
         }
       },
-      onPanEnd: (DragEndDetails details) {
+      onTapDown: (details) {
         final Size chartSize = _getChartSize();
         if (chartSize == null) {
           return;
         }
 
         final LineTouchResponse response =
-            _touchHandler?.handleTouch(FlPanEnd(Offset.zero, details.velocity), chartSize);
+            _touchHandler?.handleTouch(FlPanStart(details.localPosition), chartSize);
         if (_canHandleTouch(response, touchData)) {
+          setState(() {
```

Comment:
Why do you call `setState()` here?

Reply:
Because `onTapCancel` needs that position. Otherwise you would have the problem of needing to pass a fake `Offset.zero` which is not actually where the tap got cancelled.

Reply:
No no,
I mean ` _lastTouchedPosition = details.localPosition;` is enough,
why did you put `setState()` around it?
`setState()` leads to call build mthod again.

Reply:
Because it's a stateful variable from the previous update. Maybe I'm confused

Reply:
Just tested, you're right

Reply:
Removed setstate